### PR TITLE
Remove `built_at` for more reproducible builds

### LINKED ISSUE #113: Confirm to RFC RFC0043: Reproducible builds
To confirm to RFC0043 this buildpack should ensure that builds are reproducible. Specifically it should not include a `built_at` metadata field. In the tests that leverage this field to assert layer reuse, we should instead compare layer SHA values across rebuilds.

See also the tracking issue: https://github.com/paketo-buildpacks/rfcs/issues/165.

### PR DESCRIPTION
**Pull request by joshuatcasey**
## Summary
Remove `built_at` for more reproducible builds. Resolves #113 .

## Use Cases
See #113 

## Checklist
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by joshuatcasey** [comment]
This PR is perhaps bigger than anticipated. I included `dependency-sha` as a way to confirm that the layer is still being created as expected. We can discuss.

Also, `go generate` made some additional changes.

File: build.go [hunk 1]
```diff
@@ -51,8 +51,10 @@ func Build(entryResolver EntryResolver, installProcess InstallProcess, siteProce
 		}
 
 		logger.Process("Executing build process")
-		duration, err := clock.Measure(func() error {
-			return installProcess.Execute(context.WorkingDir, packagesLayer.Path, cacheLayer.Path)
+		var checksum string
+		duration, err := chronos.DefaultClock.Measure(func() error {
```
Comment (robdimsdale) [inline build.go:55]: In the interests of making the smallest change, I would leave the interface with an injected clock. We can have a separate conversation about the merits of removing this from the interface later.

**Review by robdimsdale — CHANGES_REQUESTED** [review]
I have one main question and a couple of small thoughts:

My main question is: Why did you create your own checksum and put that in the layer metadata vs leveraging the layer SHA itself? Specifically, do you have concerns with the approach taken in https://github.com/paketo-buildpacks/cpython/pull/322? That approach seems much simpler to me but perhaps I'm missing something.

Other thoughts:
* The re-generation of the fakes is fine - occasionally an update to the `faux` tool will result in changes to the fakes. I don't think we need to make a separate commit/PR for that, but I'm open to it if you prefer.
* I see the reasoning for removing the `chronos.Clock` from the interface, but in the interest of keeping a change small (and e.g. easy to revert) I'd suggest that we keep the existing interface. Please feel free to open an issue to discuss the refactor and we can weigh in on the pros/cons there.

File: build.go (hunk 1 above)
Comment (joshuatcasey) [reply build.go:55]: Not sure why this response was put here

Comment (joshuatcasey) [reply build.go:55]: Please disregard the previous two answers.

**Comment by joshuatcasey** [comment]
@robdimsdale regarding the SHA: buildpacks such as `cpython` and `pip` are downloading a dependency via `packit.Deliver` and that depedency has a SHA.

This buildpack is running a process `pip install` so it needs to generate its own SHA (that is, if it needs one).

**Comment by robdimsdale** [comment]
@joshuatcasey I think we are talking about two separate issues.

Occam, the integration testing framework, [gives us a way to get the layer SHA that the lifecycle populates](https://github.com/paketo-buildpacks/occam/blob/911125c5a9bf692c1135f60e718dedb46473b407/image.go#L20). We can leverage this in our integration tests (e.g. [cpython](https://github.com/paketo-buildpacks/cpython/blob/37203982566f00eab1593e1c9bd86df7930cf9c0/integration/layer_reuse_test.go#L154)).

What you have implemented would allow the buildpack to determine if its own filesystem has changed. This could be used within the buildpack to e.g. determine if we should re-use the pre-existing layer.

It might be the case that in order to re-use the layer (and hence implement this PR) we leverage something like what you've written to selectively re-use the cache layer. We will know once we implement an integration test using that occam framework. Either the integration test will Just Work™ or the scope of this issue increases significantly.

**Comment by robdimsdale** [comment]
Alright, this LGTM. I'll approve once it's rebased off main.

I'm still slightly confused as to why the layer-reuse actually works, but I can explore that on my own time.

**Comment by joshuatcasey** [comment]
@robdimsdale pushed a merge commit to bring this up to date.

**Review by robdimsdale — APPROVED** [review]

**Pull request merged**